tck/src/main/java/com/akkaserverless/javasdk/tck/model/localpersistenceeventing/LocalPersistenceSubscriber.java

```diff
@@ -18,63 +18,73 @@
-    return convert(context, cloudEvent, eventOne.getStep());
+      LocalPersistenceEventing.EventOne eventOne) {
+    return convert(actionContext(), eventOne.getStep());
   }
 
   public Source<Action.Effect<LocalPersistenceEventing.Response>, NotUsed> processEventTwo(
-      ActionContext context, CloudEvent cloudEvent, LocalPersistenceEventing.EventTwo eventTwo) {
-    return Source.from(eventTwo.getStepList()).map(step -> convert(context, cloudEvent, step));
+      LocalPersistenceEventing.EventTwo eventTwo) {
+    ActionContext context = actionContext();
+    return Source.from(eventTwo.getStepList()).map(step -> convert(context, step));
   }
 
-  public LocalPersistenceEventing.Response processAnyEvent(
-      JsonMessage jsonMessage, CloudEvent cloudEvent) {
-    return LocalPersistenceEventing.Response.newBuilder()
-        .setId(cloudEvent.subject().orElse(""))
-        .setMessage(jsonMessage.message)
-        .build();
+  public Action.Effect<LocalPersistenceEventing.Response> processAnyEvent(Any any) {
+    JsonMessage jsonMessage = actionContext().deserializeJson(JsonMessage.class, any);
+    return effects()
+        .reply(
+            LocalPersistenceEventing.Response.newBuilder()
+                .setId(actionContext().eventSubject().orElse(""))
+                .setMessage(jsonMessage.message)
+                .build());
   }
 
   public Action.Effect<LocalPersistenceEventing.Response> processValueOne(
-      ActionContext context, CloudEvent cloudEvent, LocalPersistenceEventing.ValueOne valueOne) {
-    return convert(context, cloudEvent, valueOne.getStep());
+      LocalPersistenceEventing.ValueOne valueOne) {
+    ActionContext context = actionContext();
+    return convert(context, valueOne.getStep());
   }
 
   public Source<Action.Effect<LocalPersistenceEventing.Response>, NotUsed> processValueTwo(
-      ActionContext context, CloudEvent cloudEvent, LocalPersistenceEventing.ValueTwo valueTwo) {
-    return Source.from(valueTwo.getStepList()).map(step -> convert(context, cloudEvent, step));
+      LocalPersistenceEventing.ValueTwo valueTwo) {
+    ActionContext context = actionContext();
+    return Source.from(valueTwo.getStepList()).map(step -> convert(context, step));
   }
 
-  public LocalPersistenceEventing.Response processAnyValue(
-      JsonMessage jsonMessage, CloudEvent cloudEvent) {
-    return LocalPersistenceEventing.Response.newBuilder()
-        .setId(cloudEvent.subject().orElse(""))
-        .setMessage(jsonMessage.message)
-        .build();
+  public Effect<LocalPersistenceEventing.Response> processAnyValue(Any any) {
+    JsonMessage jsonMessage = actionContext().deserializeJson(JsonMessage.class, any);
+    return effects()
+        .reply(
+            LocalPersistenceEventing.Response.newBuilder()
+                .setId(actionContext().eventSubject().orElse(""))
+                .setMessage(jsonMessage.message)
+                .build());
```

Comment:
did we have support for encoding JSON also? Like publishing a JSON message to a topic?

Would that be similar with `Any` as the response type and then explicit conversion?

Reply:
We already have `emitJson`-effects in the value and event sourced entities, so I guess we can align with that. Not entirely sure I understand how publishing to a topic works and if that fits though, will look into that.

Reply:
publishing to a topic is the returned message when proto defined with eventing.out

Reply:
Sorry, I was wrong about that, `emitJson` was TCK action methods. After some more investigation any parameter or return value where you instead used a non proto Java class annotated with `@Jsonable` would be automagically serialized/deserialized into a proto any with the akka serverless json type_url.